moto/sagemaker/responses.py

```diff
@@ -0,0 +1,136 @@
+            response = {
+                "NotebookInstanceArn": sagemaker_notebook.arn,
+            }
+            return 200, {}, json.dumps(response)
+        except AWSError as err:
+            return err.response()
+
+    @amzn_request_id
+    def describe_notebook_instance(self):
+        notebook_instance_name = self._get_param("NotebookInstanceName")
+        try:
+            notebook_instance = self.sagemaker_backend.get_notebook_instance(
+                notebook_instance_name
+            )
+            response = {
+                "NotebookInstanceArn": notebook_instance.arn,
+                "NotebookInstanceName": notebook_instance.notebook_instance_name,
+                "NotebookInstanceStatus": notebook_instance.status,
+                "Url": notebook_instance.url,
+                "InstanceType": notebook_instance.instance_type,
+                "SubnetId": notebook_instance.subnet_id,
+                "SecurityGroups": notebook_instance.security_group_ids,
+                "RoleArn": notebook_instance.role_arn,
+                "KmsKeyId": notebook_instance.kms_key_id,
+                # ToDo: NetworkInterfaceId
+                "LastModifiedTime": str(notebook_instance.last_modified_time),
+                "CreationTime": str(notebook_instance.creation_time),
+                "NotebookInstanceLifecycleConfigName": notebook_instance.lifecycle_config_name,
+                "DirectInternetAccess": notebook_instance.direct_internet_access,
+                "VolumeSizeInGB": notebook_instance.volume_size_in_gb,
+                "AcceleratorTypes": notebook_instance.accelerator_types,
+                "DefaultCodeRepository": notebook_instance.default_code_repository,
+                "AdditionalCodeRepositories": notebook_instance.additional_code_repositories,
+                "RootAccess": notebook_instance.root_access,
+            }
+            return 200, {}, json.dumps(response)
+        except AWSError as err:
+            return err.response()
+
+    @amzn_request_id
+    def start_notebook_instance(self):
+        notebook_instance_name = self._get_param("NotebookInstanceName")
+        notebook_instance = self.sagemaker_backend.get_notebook_instance(
+            notebook_instance_name
+        )
+        notebook_instance.start()
+        return 200, {}, json.dumps("{}")
+
+    @amzn_request_id
+    def stop_notebook_instance(self):
+        notebook_instance_name = self._get_param("NotebookInstanceName")
+        notebook_instance = self.sagemaker_backend.get_notebook_instance(
+            notebook_instance_name
+        )
+        notebook_instance.stop()
+        return 200, {}, json.dumps("{}")
+
+    @amzn_request_id
+    def delete_notebook_instance(self):
+        notebook_instance_name = self._get_param("NotebookInstanceName")
+        self.sagemaker_backend.delete_notebook_instance(notebook_instance_name)
+        return 200, {}, json.dumps("{}")
+
+    @amzn_request_id
+    def list_tags(self):
```

Comment:
Same here actually, this can be extracted as well - the Responses-class shouldn't know that how tags are stored

Reply:
Done.